src/Condition.js

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2019 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+const url = require('url');
+
+// To avoid forward referencing the transformer function
+let transform;
+
+const Affix = {
+  PREFIX: 1,
+  INFIX: 2,
+};
+
+/**
+ * Boolean conditions
+ */
+const booleanConditionMap = {
+  or: {
+    name: 'or',
+    type: Affix.INFIX,
+    op: ' || ',
+  },
+  and: {
+    name: 'and',
+    type: Affix.INFIX,
+    op: ' && ',
+  },
+  not: {
+    name: 'not',
+    type: Affix.PREFIX,
+    op: ' !',
+  },
+};
+
+class BooleanCondition {
+  constructor(condition, children) {
+    this._condition = condition;
+
+    switch (condition.type) {
+      case Affix.PREFIX:
+        this._children = transform(children);
+        break;
+      default:
+        this._children = children.map(child => transform(child));
+        break;
+    }
+  }
+
+  toVCL() {
+    let clause;
+
+    switch (this._condition.type) {
+      case Affix.PREFIX:
+        clause = this._children.toVCL();
+        return `${this._condition.op}(${clause})`;
+      default:
+        clause = this._children.map(child => child.toVCL()).join(this._condition.op);
+        return `(${clause})`;
+    }
+  }
+
+  evaluate(req) {
+    switch (this._condition.name) {
+      case 'and':
+        return this._children.reduce((result, child) => result && child.evaluate(req), true);
+      case 'or':
+        return this._children.reduce((result, child) => result || child.evaluate(req), false);
+      default:
```

Comment:
would probably be more robust to have:

```
case 'not':
  ...
default:
   throw Error('unexpected condition name')

Reply:
It would, but then I lose my 100% code coverage :wink:

Reply:
> It would, but then I lose my 100% code coverage 😉

then you need to add the respective test....

Reply:
Transformed it to function lookup, so the problem went away 😄